### PR TITLE
OCPQE-14434: hypershift nodepool upgrade

### PR DESCRIPTION
**Pull request by LiangquanLi930**
https://issues.redhat.com//browse/OCPQE-14434

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-cp-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-ec-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-worker-node-p1-f4

**Comment by LiangquanLi930** [comment]
@kuiwang02 Could you help to review this PR? thanks

File: ci-operator/step-registry/cucushift/upgrade/hypershift/worker-node/toimage/cucushift-upgrade-hypershift-worker-node-toimage-commands.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+function health_check() {
+    echo "Step #1: Check all cluster operators get stable and ready"
+    timeout 900s bash <<EOT
+until
+  oc wait clusteroperators --all --for='condition=Available=True' --timeout=30s && \
+  oc wait clusteroperators --all --for='condition=Progressing=False' --timeout=30s && \
+  oc wait clusteroperators --all --for='condition=Degraded=False' --timeout=30s;
+do
+  sleep 30 && echo "Cluster Operator ingress Degraded=True,Progressing=True,or Available=False";
```
Comment (shellyyang1989) [inline ci-operator/step-registry/cucushift/upgrade/hypershift/worker-node/toimage/cucushift-upgrade-hypershift-worker-node-toimage-commands.sh:72]: Just a minor question: why we report ingress degraded here?

Comment (LiangquanLi930) [reply ci-operator/step-registry/cucushift/upgrade/hypershift/worker-node/toimage/cucushift-upgrade-hypershift-worker-node-toimage-commands.sh:72]: sorry, let me update^^

**Comment by shellyyang1989** [comment]
lgtm

**Comment by kuiwang02** [comment]
Iamok from golang's view

**Comment by LiangquanLi930** [comment]
@liangxia Could you help to review this PR? thanks

**Pull request merged**